includes/handler/class-inbox.php

```diff
@@ -59,23 +86,39 @@ public static function handle_inbox_requests( $data, $user_id, $type, $activity
 			$object_types = \array_map( 'Activitypub\camel_to_snake_case', $object_types );
 
 			if ( is_array( $data['object'] ) && ( empty( $data['object']['type'] ) || ! \in_array( \strtolower( $data['object']['type'] ), $object_types, true ) ) ) {
-				$success = false;
-				$id      = new \WP_Error( 'activitypub_inbox_ignored', 'Activity type not configured to be persisted in inbox.' );
+				$inbox_id = new \WP_Error( 'activitypub_inbox_ignored', 'Activity type not configured to be persisted in inbox.' );
 			}
 		}
 
-		if ( $success ) {
-			$id = Inbox_Collection::add( $activity, $user_id );
+		if ( ! \is_wp_error( $inbox_id ) ) {
+			// Pass user_id as-is (can be int or array) to Inbox_Collection::add().
+			$inbox_id = Inbox_Collection::add( $activity, $user_ids );
 		}
 
+		// Normalize user_id to array for action hooks.
+		$user_ids = is_array( $user_ids ) ? $user_ids : array( $user_ids );
```

Comment:
```suggestion
		$user_ids = (array) $user_ids;
```
This might be good to do at the top of the function so `Inbox_Collection::add()` can take advantage of it, too.